contracts/loot/ItemConstants.cairo

```diff
@@ -0,0 +1,746 @@
+# Item Structs
+#   A struct that holds the Realm statistics.
+#   Each module will need to add a struct with their metadata.
+#
+# MIT License
+
+%lang starknet
+
+struct Item:
+    member Id : felt  # item id 1 - 100
+    member Class : felt  # loot item class
+    member Slot : felt  # head, chest, waist, feet, hands, neck, ring, weapon
+    member Agility : felt  # Computed Agility of the item
+    member Attack : felt  # Computed Attack of the item
+    member Armour : felt  # Computed Armour of the item
+    member Wisdom : felt  # Computed Wisdom of the item
+    member Vitality : felt  # Computed Vitality of the item
+    member Prefix : felt  # Stored value if item has a Prefix
+    member Suffix : felt  # Stored value if item has a Suffix
+    member Order : felt  # Stored value if item has a Order
+    member Bonus : felt  # Stored value if item has a Bonus
+    member Level : felt  # Stored value if item has a Level
+    member Age : felt  # Timestamp of when item was created
+    member XP : felt  # accured XP
+end
+
+struct Adventurer:
+    member Class : felt
+
+    # computed of all items stats
+    member Agility : felt
+    member Attack : felt
+    member Armour : felt
+    member Wisdom : felt
+    member Vitality : felt
+
+    # store item NFT id when equiped
+    member Neck : Item
+    member Weapon : Item
+    member Ring : Item
+    member Chest : Item
+    member Head : Item
+    member Waist : Item
+    member Feet : Item
+    member Hands : Item
+
+    # other unique state
+    member Age : felt
+    member Name : felt  # mint time
+    member XP : felt
+    member Order : felt
+end
+
+struct AdventurerState:
+    # other unique state p1
+    member Class : felt
+    member Age : felt
+    member Name : felt  # mint time
+    member XP : felt
+    member Order : felt
+
+    # store item NFT id when equiped
+    # Packed Stats p2
+    member NeckId : felt
+    member WeaponId : felt
+    member RingId : felt
+    member ChestId : felt
+
+    # Packed Stats p3
+    member HeadId : felt
+    member WaistId : felt
+    member FeetId : felt
+    member HandsId : felt
+end
+
+struct PackedAdventurerStats:
+    member p1 : felt
+    member p2 : felt
+    member p3 : felt
+end
+
+namespace Class:
+    const All = 1
+    const Mage = 2
+    const Ranger = 3
+    const Warrior = 4
+end
+
+namespace ItemClass:
+    const Pendant = Class.All
+    const Necklace = Class.All
+    const Amulet = Class.All
+    const SilverRing = Class.All
+    const BronzeRing = Class.All
+    const PlatinumRing = Class.All
+    const TitaniumRing = Class.All
+    const GoldRing = Class.All
+    # TODO: add
+end
+
+namespace Slot:
+    const Weapon = 1
+    const Chest = 2
+    const Head = 3
+    const Waist = 4
+    const Feet = 5
+    const Hands = 6
+    const Neck = 7
+    const Ring = 8
+end
```

Comment:
Just a random comment - without checking how this is used, it looks like an ideal candidate for bitmap values, so they can be ANDed / ORed together:

```
const Weapon = 0x1
const Chest = 0x10
const Head = 0x100
# etc.
```